packages/@react-native-windows/cli/src/runWindows/runWindows.ts

```diff
@@ -15,16 +57,61 @@ import {Command, Config} from '@react-native-community/cli-types';
 import {runWindowsOptions, RunWindowsOptions} from './runWindowsOptions';
 
 import {autoLinkCommand} from './utils/autolink';
+import {totalmem, cpus} from 'os';
+import {execSync} from 'child_process';
+
+function getDiskFreeSpace(drivePath: string | null): number {
+  const out = execSync(`dir /-C ${drivePath}`)
+    .toString()
+    .split('\r\n');
+  const line = out[out.length - 2];
+  const result = line.match(/(\d+) [^\d]+(\d+) /);
+  if (result && result.length > 2) {
+    return Number(result[2]);
+  }
+  return -1;
+}
 
-function ExitProcessWithError(loggingWasEnabled: boolean): never {
+function SetExitProcessWithError(loggingWasEnabled: boolean): void {
   if (!loggingWasEnabled) {
     console.log(
       `Re-run the command with ${chalk.bold('--logging')} for more information`,
     );
   }
-  process.exit(1);
+  process.exitCode = 1;
+}
+
+function isMSFTInternal(): boolean {
+  return (
+    process.env.USERDNSDOMAIN !== undefined &&
+    process.env.USERDNSDOMAIN.endsWith('microsoft.com')
```

Comment:
Should we ensure `microsoft` is the full name by using a '.' in front?
```suggestion
    process.env.USERDNSDOMAIN.endsWith('.microsoft.com')
```